code/test-utils/src/main/java/com/adobe/marketing/tester/util/JSONObjectAsserts.kt

```diff
@@ -0,0 +1,940 @@
+    }
+
+    /**
+     * Constructs a nested mutable map structure based on the provided path, with the deepest nested key pointing to the given `pathString`.
+     *
+     * For instance, given a path of `["a", "b", "c"]` and a `pathString` of "value", the resulting map would be:
+     * `{"a": {"b": {"c": "value"}}}`.
+     *
+     * @param path A mutable list of strings representing the sequence of nested keys for the map structure.
+     * @param pathString The string value that will be associated with the deepest nested key in the constructed map.
+     *
+     * @return A mutable map representing the nested structure constructed from the `path`, with the deepest nested key pointing to `pathString`.
+     */
+    private fun construct(path: MutableList<String>, pathString: String): MutableMap<String, Any> {
+        if (path.isEmpty()) return mutableMapOf()
+        val first = path.removeAt(0)
+        return if (path.isEmpty()) {
+            mutableMapOf(first to pathString)
+        } else {
+            mutableMapOf(first to construct(path, pathString))
+        }
+    }
+
+    private fun generatePathTree(paths: List<String>): Map<String, Any>? {
+        // Matches array subscripts, capturing the brackets and inner content (e.g., "[123]", "[*123]").
+        val arrayIndexRegex = """(\[.*?\])"""
+        val tree: MutableMap<String, Any> = mutableMapOf()
+
+        for (exactValuePath in paths) {
+            var allPathComponents: MutableList<String> = mutableListOf()
+
+            // Break the path string into its individual components.
+            val keyPathComponents = getKeyPathComponents(exactValuePath)
+            for (pathComponent in keyPathComponents) {
+                // Convert any escaped periods to actual periods, because the actual key names with periods in the JSON
+                // will not be escaped
+                val pathComponent = pathComponent.replace("\\.", ".")
+
+                // Extract all array access levels from the pathComponent, if they exist.
+                // Note: This regex extracts sequences of square brackets and their contents ("[___]").
+                // However, positions of these brackets within the path component are not considered (e.g., "key0[2]key1[3]" will yield "key0" and "[2][3]").
+                val arrayComponents = getCapturedRegexGroups(pathComponent, arrayIndexRegex)
+
+                // If no array subscripts are detected, directly add the entire path component.
+                if (arrayComponents.isEmpty()) {
+                    allPathComponents.add(pathComponent)
+                }
+                // If array subscripts are detected, extract the path preceding the first subscript.
+                else {
+                    val bracketIndex = pathComponent.indexOf("[")
+                    if (bracketIndex == -1) {
+                        println("TEST ERROR: unable to get bracket position from path: $pathComponent. Skipping exact path: $exactValuePath")
```

Comment:
in ios we seem to assert fail and break when we detect this case, any reason why we continue here?